### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by ZuluPro**
Improved documentation for v3 with:
- Upgrade manual
- Database connectors
- Storage configuration
- Integration

### CODE REVIEW

**Comment by ZuluPro** [comment]
@benjaoming @mjs7231 
What do you think about ?

File: requirements-docs.txt [hunk 1]
```diff
@@ -1 +1,3 @@
 Sphinx==1.3.1
+sphinx-django-command
```
Comment (benjaoming) [inline requirements-docs.txt]: nice, I didn't know this -- thanks for adding it and simplifying / improving the docs :)

Comment (benjaoming) [inline requirements-docs.txt]: Btw. it's possible to use sphinx-apidoc to build docs and include them like this:

```

.. automodule:: dbbackup.settings
  :undoc-members:
  :show-inheritance:

Comment (benjaoming) [inline requirements-docs.txt]: ...you can annotate a setting like this:

```
#: Fake host
HOSTNAME = getattr(settings, 'DBBACKUP_HOSTNAME', socket.gethostname())
```

Comment (ZuluPro) [inline requirements-docs.txt]: :smile: 
I created sphinx-django-command especially for improve DBBackup docs.
And other projects, of course.

Comment (benjaoming) [inline requirements-docs.txt]: I'll have a look at it! Thanks and congrats :) In other django-apps, I'm using a dummy django project to have the django stack loaded such that for instances `models.py` can be introspected, for instance here: https://github.com/learningequality/kolibri/blob/master/docs/conf.py#L37

**Pull request merged**